### ISSUE TITLE
Empty chat dialog

### ISSUE THREAD

**Issue by BiTOk** [psi-im/psi#345]
After update to Psi v1.3-dev (2018-02-20, webengine) from psi-plus-1.2.154 (archlinux) I get empty chat dialog. Chat dialog lost webview. In console I get: https://gist.github.com/BiTOk/f29c6c6710c352f34b1b22185561552e.

**Comment by Ri0n** [comment]
Qt 5.10.1?

**Comment by BiTOk** [comment]
Yes, 5.10.1 x64.

**Comment by Ri0n** [comment]
Then it's something known. Ask in chat @Vitozz (KuKuRuZo)

**Comment by Ri0n** [comment]
I see some strange webkit related problems too. particularly with contact list (it's not pulated) with Adium themes. Pretty strange..

**Comment by vbooka1** [comment]
The issue still persists in 2022 lol, at least in one distro: openSUSE has broken Qt Webengine so psi-plus has empty chat dialog, regardless if installed from repositories (both Leap and Tumbleweed) or built from the source with -DCHAT_TYPE=WEBENGINE.

You need to build psi-plus with -DCHAT_TYPE=WEBKIT to make chats work.

**Comment by tehnick** [comment]
@vbooka1 
Make sure that `QtWebEngineProcess` program is present in your system:
```
$ find /usr/lib /usr/bin | grep -i QtWebEngineProcess
/usr/lib/x86_64-linux-gnu/qt5/libexec/QtWebEngineProcess
```

In Debian and Ubuntu it is packaged into `libqt5webenginecore5` package.

**Comment by vbooka1** [comment]
it is 

```
$ file /usr/lib64/qt5/libexec/QtWebEngineProcess
/usr/lib64/qt5/libexec/QtWebEngineProcess: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=dc2e96c710ae092ff34ce2c20244e0833eed8a20, for GNU/Linux 3.2.0, stripped
```

**Comment by Ri0n** [comment]
If it's a problem with JS incomplete initialization then being started from terminal it may show some useful log messages related to the problem.

**Comment by tehnick** [comment]
@vbooka1 
Could you show a screenshot of such empty window to make sure that we are talking about the same bug?

**Comment by Ri0n** [comment]
@tehnick If I understand correctly some threads controlling webengine are crashing.
I don't have any explanation for this. IIRC all suspicious places were already patched.

**Comment by Ri0n** [comment]
Maybe /usr/lib64/qt5/libexec/QtWebEngineProcess is somewhat broken. Maybe Qt can't properly handle it's early disruption and crashes itself.
Just a wild guess.

**Comment by Ri0n** [comment]
> openSUSE has broken Qt Webengine

alright. likely true.

**Comment by tehnick** [comment]
> /usr/lib64/qt5/libexec/QtWebEngineProcess

It would be nice to see output of:
```
ldd /usr/lib/x86_64-linux-gnu/qt5/libexec/QtWebEngineProcess
```

Also Qt may expect to find `QtWebEngineProcess` in the another place. And this may be checked only using `trace` command line tool.

**Comment by vbooka1** [comment]
@tehnick 

> @vbooka1 Could you show a screenshot of such empty window to make sure that we are talking about the same bug?

the message input field has white background, the chat field has gray background and has no any text after I send messages.

![scr](https://user-images.githubusercontent.com/29492874/168649171-fa47fa16-4a6f-43d7-8d53-b2c6d2718be2.png)

`ldd /usr/lib64/qt5/libexec/QtWebEngineProcess` : https://gist.github.com/vbooka1/cc5f4d97f044a995fc821e69e5b3a5f3

`ldd /usr/bin/psi-plus` : https://gist.github.com/vbooka1/b410870315ecbb680db993537851362e

running `/usr/bin/psi-plus` : https://gist.github.com/vbooka1/e833fb394b279901d343157d50120dda

partial `strace` without private and unnecessary data such as `env` contents, ELF files binary content, etc:  https://gist.github.com/vbooka1/9f38b816efa153c6977244e0dc5eef11

take a look at this line:
  
`[pid 215353] access("/usr/lib64/qt5/plugins/webengine", F_OK) = -1 ENOENT (No such file or directory) <0.000059>`

there is no such directory indeed, however there is another interesting one:

```
$ ls -la /usr/lib64/qt5/plugins/|grep web
-rwxr-xr-x  1 root root   43688 Aug 26  2021 kcm_webshortcuts.so
-rwxr-xr-x  1 root root   27504 Aug 27  2021 krunner_webshortcuts.so
drwxr-xr-x  2 root root    4096 May 15 04:00 webview
$ ls -la /usr/lib64/qt5/plugins/webview/
total 60
drwxr-xr-x  2 root root  4096 May 15 04:00 .
drwxr-xr-x 60 root root 12288 May 16 13:06 ..
-rwxr-xr-x  1 root root 41672 Apr 24  2021 libqtwebview_webengine.so
```

**Comment by vbooka1** [comment]
> I think it's necessary to install debug packages for qtwebengine package and take a look at traces once again.

sorry, I'm too lazy for this, especially when there is a solution already:  https://github.com/psi-im/psi/issues/345#issuecomment-1126908814 - I've compiled Psi+ from source with Webkit and it works well.
Also there is extremely small amount of affected users - it was the first comment in 3 years!

> Also you don't need to open any chat dialog to load web theme.

I thought that the `[20220516 19:51:34] ChatView theme is not loaded. this is fatal (unknown:0, unknown)` message might be important, because it appears only when I open a chat window (i.e. when I trigger the bug)

**Comment by Ri0n** [comment]
> sorry, I'm too lazy for this

Ok Ok :)

> I thought ...

Psi starts hidden webengine on start to init theme resources (loads theme) and it's where it fails. 
The message just means it can't properly open chat dialog without loaded theme in advance.

**Comment by tehnick** [comment]
@vbooka1 

> the chat field has gray background and has no any text after I send messages

Yep, I have faced with very similar bug in macOS. It was caused by features of Qt build in Homebrew: QtWebEngineProcess executable used absolute path to Qt libs and did not worked with libs from app bundle.

> there is no such directory indeed

It does not needed for proper work of QtWebEngine.

> running /usr/bin/psi-plus ...
> partial strace ...

Thank you for such detailed logs! Unfortunately they look normal. I do not have any new ideas what may cause this problem.
